### PR TITLE
Add provider to create payment

### PR DESCRIPTION
**Pull request by brycekahle**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -280,7 +281,7 @@ export default class GoCommerce {
       }));
     } else {
       return Promise.reject(
-        "Invalid paymentDetails - must have an order_id, an amount and a stripe_token or a paypal_payment_id and paypal_user_id"
+        "Invalid paymentDetails - must have an order_id, an amount, a provider, and a stripe_token or a paypal_payment_id and paypal_user_id"
```
Comment (rybit) [inline src/index.js:284]: Instead of doing this couldn't you give better error messages?

Comment (brycekahle) [reply src/index.js:284]: It depends on how much we want to do in the client. It is possible to attempt to use PayPal, but not have it enabled on the server side.

Comment (rybit) [reply src/index.js:284]: I was thinking just doing the same validation that you do in the server. Based on the provider you check that the right values are set. 

Also, I am comfortable with just delegating that down to the server.

Comment (brycekahle) [reply src/index.js:284]: Definitely a design decision. I'm on the fence about having the validation on the client-side (beyond the common parameters). With two providers, it is manageable, but I could see it getting complex with more.

Comment (rybit) [reply src/index.js:284]: Let's leave it out. It is simple this way and I don't want to spend forever thinking about it :) We can make the lib smarter if we need to later

**Review by rybit — APPROVED** [review]

**Comment by rybit** [comment]
lgtm

**Pull request merged**